metron-platform/metron-common/src/main/java/org/apache/metron/common/query/PredicateProcessor.java

```diff
@@ -18,12 +18,20 @@
 
 package org.apache.metron.common.query;
 
-import org.antlr.v4.runtime.*;
-import org.apache.metron.common.query.generated.*;
+import org.antlr.v4.runtime.ANTLRInputStream;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.TokenStream;
+import org.apache.metron.common.query.generated.PredicateLexer;
+import org.apache.metron.common.query.generated.PredicateParser;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
 
 
 public class PredicateProcessor {
   public boolean parse(String rule, VariableResolver resolver) {
+    if (isEmpty(rule)) {
```

Comment:
That works. || or && are equivalent in this situation
On Thu, May 12, 2016 at 18:56 Michael Miklavcic <notifications@github.com>
wrote:

> In
> metron-platform/metron-common/src/main/java/org/apache/metron/common/query/PredicateProcessor.java
> <https://github.com/apache/incubator-metron/pull/119#discussion_r63111019>
> :
>
> >
> >
> >  public class PredicateProcessor {
> >    public boolean parse(String rule, VariableResolver resolver) {
> > +    if (isEmpty(rule)) {
>
> On second thought, what about rule == null || isEmpty(rule.trim())? I
> can't think of a reason to handle null different from "" or " ". The trim
> would only be evaluated if rule != null, and now we're also covering the
> null condition. Thoughts?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-metron/pull/119/files/b3af68484718a66393f4b0417bc525c462afed5d#r63111019>
>